tests/plugins/wysiwygarea/detached.js

```diff
@@ -0,0 +1,206 @@
+/* bender-tags: editor, 4462 */
+/* bender-ckeditor-plugins: basicstyles,toolbar */
+
+( function() {
+	'use strict';
+
+	function isSupportedEnvironment() {
+		// Skip IE's below version 11. They don't support MutationObserver.
+		return !( CKEDITOR.env.ie && CKEDITOR.env.version < 11 );
+	}
+
+	var startupData = '<p>CKEditor4</p>';
+
+	bender.test( {
+		'test reattached editor contains the same data with observed default dom object': function() {
+			if ( !isSupportedEnvironment() ) {
+				assert.ignore();
+			}
+
+			bender.editorBot.create( {
+				startupData: startupData,
+				name: 'editor' + new Date().getTime()
+			}, function( bot ) {
+				var editorContainer = bot.editor.container,
+					editorContainerParent = editorContainer.getParent();
+
+				editorContainer.remove();
+				editorContainerParent.append( editorContainer );
+
+				timeoutAssert( bot, startupData, 'Reattached editor should have the same data. Observing default object.' );
+
+				wait();
+			} );
+		},
+
+		'test reattached editor parent restores editor data with observed default dom object': function() {
+			if ( !isSupportedEnvironment() ) {
+				assert.ignore();
+			}
+
+			bender.editorBot.create( {
+				startupData: startupData,
+				name: 'editor' + new Date().getTime()
+			}, function( bot ) {
+				var editorContainerParent = bot.editor.container.getParent(),
+					parentParent = editorContainerParent.getParent();
+
+				editorContainerParent.remove();
+				parentParent.append( editorContainerParent );
+
+				timeoutAssert( bot, startupData, 'Reattached editor parent should restore editor data. Observing default object.' );
+
+				wait();
+			} );
+		},
+
+		'test reattached editor restores editor data with editor parent observed': function() {
+			if ( !isSupportedEnvironment() ) {
+				assert.ignore();
+			}
+
+			var observableParent = CKEDITOR.document.getById( 'editorDetachableParent' );
+
+			bender.editorBot.create( {
+				name: 'editor',
+				startupData: startupData,
```

Comment:
We use `startupData` in assertions. That means, if we put it in `.html` file, we need to add additional data extraction from HTML. I would stay with the current solution. Actually, current HTML is provided for hierarchy visualization.

Reply:
Yeah, we have to keep `startupData` variable for assertions, but I don't see why we'd need to extract data from HTML?

Reply:
I agreed with @sculpt0r that having the data in variable is less prone to produce errors in case of changes in the expected data or test structure, so we'll leave it as is 👍🏻